src/Core/Content/Flow/Dispatching/Action/SetOrderStateAction.php

```diff
@@ -218,4 +223,15 @@ private function getStateMachineId(string $machine): ?string
 
         return $id ?: null;
     }
+
+    private function getMachineIdFromOrderTransaction(string $orderId): ?string
+    {
+        return $this->connection->fetchOne(
+            'SELECT LOWER(HEX(id)) FROM ' . self::ORDER_TRANSACTION . ' WHERE order_id = :id AND version_id = :version ORDER BY created_at DESC',
```

Comment:
definitly not worth a constant
```suggestion
            'SELECT LOWER(HEX(id)) FROM `order_transaction` WHERE order_id = :id AND version_id = :version ORDER BY created_at DESC',
```

Reply:
self::ORDER_TRANSACTION is used 5 times. I would keep it